config/config.php

```diff
@@ -20,6 +20,10 @@
     \Mezzio\ConfigProvider::class,
     \Mezzio\Router\ConfigProvider::class,
 
+    class_exists(\Laminas\Diactoros\ConfigProvider::class)
+        ? \Laminas\Diactoros\ConfigProvider::class
+        : function(){ return[]; },
+
```

Comment:
```suggestion
    \Laminas\Diactoros\ConfigProvider::class,

```

This class will *always* exist during an initial installation; the only time it won't exist is if the user has removed Diactoros from their application. Since calling `composer remove laminas/laminas-diactoros` would remove the entry, it's safe for us to assume the class exists, and just list it as a provider.

(The only reason we have the `class_exists()` check for Swoole's config provider is because it is NOT installed by default, and the package is not listed as a config provider. This is to allow using different approaches in production versus development — but we also wanted a  turn-key way to enable it if you add the package to your application later. so we used the `class_exists` construct to do so here.)

Reply:
in composer, it defines `"laminas/laminas-diactoros": "^1.8.7 || ^2.3.0",`, is the config provider cherry-picked to v1 as well?

Reply:
There's no reason for us to pin to the 1.8 series any more; for a new minor release, I'd just remove that constraint (which I've noted in another comment now).

Reply:
ok, implemented.